ds3-sdk-samples/src/main/java/com/spectralogic/ds3client/samples/Ds3ServiceListExample.java

```diff
@@ -40,6 +42,27 @@ public static void main(final String args[]) throws IOException, SignatureExcept
             for (final Ds3Bucket bucket : response.getListAllMyBucketsResult().getBuckets()) {
                 System.out.println(bucket.getName());
             }
-        }
+        // Catch unknown host exceptions.    
+        } catch (final UnknownHostException e) {
+        	
+        	System.out.println("Invalid Endpoint Server Name or IP Address");
+        	
+        // Catch failed requests with unexpected status codes.
+        } catch (final FailedRequestException e) {
+        	
+        	// If this is invalid authorization.
+        	if (e.getStatusCode() == 403) {
+        		
+        		System.out.println("Invalid Access ID or Secret Key");
+        	
+        	// Else unexpected status code.
+        	} else {
+        		
+        		System.out.println("BlackPearl return an unexpected status code we did not expect");
+                // e.getStatusCode() can be used to get the status code BlackPearl returned for more accurate error handling and detection
+        		
+        	}
+                
+        } 
```

Comment:
It might be good to still show the IOException handling at the end to demonstrate the difference between FailedRequestException and IOExcepton.  Though probably not explicitly required.